pkg/vmserver/vmserver.go

```diff
@@ -144,6 +154,55 @@ func (vms VMServer) GetVirtualMachineById(id string) (hfv1.VirtualMachine, error
 
 }
 
+func (vms VMServer) GetSharedVirtualMachinesFunc(w http.ResponseWriter, r *http.Request) {
+	_, err := vms.auth.AuthN(w, r)
+	if err != nil {
+		util.ReturnHTTPMessage(w, r, 403, "forbidden", "no access to get vm")
+		return
+	}
+
+	vars := mux.Vars(r)
+
+	accessCode := vars["access_code"]
+
+	if len(accessCode) == 0 {
+		util.ReturnHTTPMessage(w, r, 500, "error", "no accessCode id passed in")
+		return
+	}
+
+	accessCodeResource, err := vms.acClient.GetAccessCode(accessCode)
```

Comment:
After merging master refactor this to acClient.GetAccessCodeWithOTAC as we changed the acClient to allow OTACs